includes/Checker/Checks/Plugin_Updater_Check.php

```diff
@@ -0,0 +1,187 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Checks\Plugin_Updater_Check
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker\Checks;
+
+use Exception;
+use WordPress\Plugin_Check\Checker\Check_Result;
+
+/**
+ * Check to detect plugin updater.
+ *
+ * @since n.e.x.t
+ */
+class Plugin_Updater_Check extends Abstract_File_Check {
+
+	/**
+	 * Amends the given result by running the check on the given list of files.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check result to amend, including the plugin context to check.
+	 * @param array        $files  List of absolute file paths.
+	 *
+	 * @throws Exception Thrown when the check fails with a critical error (unrelated to any errors detected as part of
+	 *                   the check).
+	 */
+	protected function check_files( Check_Result $result, array $files ) {
+		$php_files = self::filter_files_by_extension( $files, 'php' );
+
+		// Looks for "UpdateURI" in plugin header.
+		$this->look_for_update_uri_header( $result );
+
+		// Looks for special updater file.
+		$this->look_for_updater_file( $result, $php_files );
+
+		// Looks for plugin updater code in plugin files.
+		$this->look_for_plugin_updaters( $result, $php_files );
+
+		// Looks for plugin updater routines in plugin files.
+		$this->look_for_updater_routines( $result, $php_files );
+	}
+
+	/**
+	 * Looks for UpdateURI in plugin header and amends the given result with an error if found.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check result to amend, including the plugin context to check.
+	 */
+	protected function look_for_update_uri_header( Check_Result $result ) {
+		if ( ! function_exists( 'get_plugin_data' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/plugin.php';
+		}
+
+		$plugin_main_file = $result->plugin()->abspath();
+		$plugin_header    = get_plugin_data( $plugin_main_file );
+		if ( ! empty( $plugin_header['UpdateURI'] ) ) {
+			$this->add_result_error_for_file(
+				$result,
+				$plugin_main_file,
+				__( 'Plugin Updater detected. Use of the Update URI header is not helpful in plugins hosted on WordPress.org.', 'plugin-check' )
+			);
+		}
+	}
+
+	/**
+	 * Looks for plugin updater file and amends the given result with an error if found.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result    The check result to amend, including the plugin context to check.
+	 * @param array        $php_files List of absolute PHP file paths.
+	 */
+	protected function look_for_updater_file( Check_Result $result, array $php_files ) {
+
+		$filenames     = array_map( 'strtolower', array_map( 'basename', $php_files ) );
+		$blocked_files = array( 'plugin-update-checker.php' );
+		$matches       = array_intersect( $filenames, $blocked_files );
+
```

Comment:
I think it would be more efficient to use the `filter_files_by_regex()` method in the base class here instead of the several array functions. More importantly, the code here has the problem that it only shows the basename of the file in the error message instead of the full file name including path. I know it's like that in the original code too, but it isn't right there either.

I think you can do something like:
```php
$found_files = self::filter_files_by_regex( $php_files, '/plugin-update-checker\.php$/' )
```

If the result is not empty, it will include the full file paths.

Reply:
Thanks for feedback. Addressed in https://github.com/10up/plugin-check/pull/189/commits/e1572f3ffd07a9aeba3463cc7b981cdc2b386ead